test/GenericsSpec.hs

```diff
@@ -0,0 +1,98 @@
+module GenericsSpec where
+
+import Control.Monad.Reader
+import Data.Attoparsec.ByteString.Char8
+import Test.Hspec hiding (Arg)
+import Text.PrettyPrint.HughesPJ (Doc)
+import Weft.Generics.AllTypes
+import Weft.Generics.EmptyQuery
+import Weft.Generics.Hydrate
+import Weft.Generics.PprQuery
+import Weft.Generics.PprSchema
+import Weft.Generics.QueryParser
+import Weft.Generics.Resolve
+import Weft.Generics.Schema
+import Weft.Types
+
+
+newtype Id = Id String deriving (Generic, Show, Eq, Ord)
```

Comment:
We re-use these test types a lot, should we create a mock data module and just import that?

Reply:
Actually, on second thought, don't do this in this pr. We will have crazy conflicts with my pr. After both these PRs get merged I will clean this up.

Reply:
My thought is to just have the tests be self-contained so that no future changes to testdata will break them. I don't really care about the `Id` and `Name` things; just copy and pasted because I was lazy!

Reply:
Merge away!